src/execution_algorithms/kfac.cpp

```diff
@@ -377,6 +435,331 @@ void KFAC::do_batch_end_cbs(model& model)
   }
 }
 
+// =============================================
+// Sub-grid implementation
+// =============================================
+
+void KFAC::send_recv_kfac_inputs(model& model, lbann_comm *comm){
+  //Does not support Model parallel only Data Parallel
+
+  const auto layers = model.get_layers();
+  const El::mpi::Comm & combined_comm = comm->get_combined_grid_comm();
+  const El::mpi::Comm & trainer_comm = comm->get_KFAC_comm();
+  const int comm_size = El::mpi::Size(comm->get_KFAC_comm());
+  const int comm_rank = El::mpi::Rank(comm->get_KFAC_comm());
+}
+
+
+void KFAC::sync_weights_model(model& model, lbann_comm *comm){
```

Comment:
The implementation here works, although it would be easier to iterate through the list of weights rather than to iterate through the list of layers. It would also avoid redundant communication if multiple layers share the same weights.